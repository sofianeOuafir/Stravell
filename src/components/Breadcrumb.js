import React, { Fragment } from "react";
import Link from "next/link";

const Breadcrumb = ({ links }) => (
  <div>
    {links &&
      // linkProps would be "href" and "as" props
      links.map(({ text, ...linkProps }, index) => {
        if (linkProps.href && text) {
          return (
            <Link key={index} {...linkProps}>
              <a className="text-dark-grey no-text-decoration">
                <span className="mr1">{text}</span>
                {links.length !== index + 1 && <span className="mr1">/</span>}
              </a>
            </Link>
          );
        }
      })}
  </div>
);

export default Breadcrumb;
