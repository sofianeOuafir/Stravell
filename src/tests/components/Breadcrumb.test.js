import React from "react";
import { shallow } from "enzyme";
import Breadcrumb from "../../components/Breadcrumb";

test("should render correctly when passing links props with valid link", () => {
  const links = [
    { href: "/home", text: "Home", active: true },
    { href: "/destinations", text: "Destinations", active: false }
  ];

  const wrapper = shallow(<Breadcrumb links={links} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render correctly when passing links props with link containing no text", () => {
  const links = [
    { href: "/home", text: "", active: true },
    { href: "/destinations", text: "", active: false }
  ];

  const wrapper = shallow(<Breadcrumb links={links} />);
  expect(wrapper).toMatchSnapshot();
});
