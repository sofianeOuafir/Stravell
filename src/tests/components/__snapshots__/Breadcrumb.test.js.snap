// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when passing links props with link containing no text 1`] = `<div />`;

exports[`should render correctly when passing links props with valid link 1`] = `
<div>
  <Link
    href="/home"
    key="0"
    prefetch={true}
  >
    <a
      className="breadcrumb__link"
    >
      <span
        className="breadcrumb__link-text breadcrumb__link-text--active"
      >
        Home
      </span>
      <span
        className="mr1"
      >
        /
      </span>
    </a>
  </Link>
  <Link
    href="/destinations"
    key="1"
    prefetch={true}
  >
    <a
      className="breadcrumb__link"
    >
      <span
        className="breadcrumb__link-text "
      >
        Destinations
      </span>
    </a>
  </Link>
</div>
`;
